packages/std/src/traits.rs

```diff
@@ -3,16 +3,16 @@ use std::vec::Vec;
 use crate::errors::Result;
 use crate::types::{CanonicalAddr, HumanAddr};
 
-// Extern holds all external dependencies of the contract,
-// designed to allow easy dependency injection at runtime
-#[derive(Clone)]
+/// Holds all external dependencies of the contract.
+/// Designed to allow easy dependency injection at runtime.
+/// This cannot be copied or cloned since it would behave differently
+/// for mock storages and a bridge storage in the VM.
 pub struct Extern<S: Storage, A: Api> {
```

Comment:
hmm, how was it even possible to derive `Clone` before, when `Storage` is not clonable?

Reply:
This is a rabbit hole of Rust compiler specifics. There are a bunch of open and closed Rust issues around that topic, which I hardly understand. It is very well possible, that this is a bug or known limitation in the compiler or the standard. I better give up trying to understand this.